src/main.cpp

```diff
@@ -3082,7 +3082,28 @@ bool ContextualCheckBlockHeader(const CBlockHeader& block, CValidationState& sta
 
```

Comment:
I feel like something is missing in `CheckBlockHeader`

Reply:
Please explain.

Reply:
Nevermind.